flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -306,17 +306,19 @@ public Set<FunctionIdentifier> getUserDefinedFunctions(
                         .collect(Collectors.toSet()));
 
         // add catalog functions
-        Catalog catalog = catalogManager.getCatalog(catalogName).get();
+        Optional<Catalog> catalogOpt = catalogManager.getCatalog(catalogName);
         try {
-            catalog.listFunctions(databaseName)
-                    .forEach(
-                            name ->
-                                    result.add(
-                                            FunctionIdentifier.of(
-                                                    ObjectIdentifier.of(
-                                                            catalogName, databaseName, name))));
+            if (catalogOpt.isPresent()) {
```

Comment:
At first, I didn't use it because `listFunctions` throws a checked `DatabaseNotExistException`, but after your comment I thought we can refactor it a bit more and then it makes sense with `ifPresent`.